source/slang/slang-capability.h

```diff
@@ -247,37 +245,19 @@ struct CapabilitySet
             }
             Iterator begin() const
             {
-                // first node may not have a valid atomSet or stageSet. handle the senario of: 
-                // 1. targetNode is empty
-                // 2. stage node has no valid stage sets (not empty)
-                // 3. stage node is empty
-                // 4. atomSet has no sets
                 Iterator tmp(this->context);
                 tmp.targetNode = this->context->begin();
                 if (tmp.targetNode == this->context->end())
                     return tmp;
-
                 tmp.stageNode = (*tmp.targetNode).second.shaderStageSets.begin();
-                for(;;)
+                if (tmp.stageNode == (*tmp.targetNode).second.shaderStageSets.end())
                 {
-                    while (tmp.stageNode == (*tmp.targetNode).second.shaderStageSets.end())
-                    {
-                        tmp.targetNode++;
-                        if (tmp.targetNode == this->context->end())
-                            return tmp;
-                        tmp.stageNode = (*tmp.targetNode).second.shaderStageSets.begin();
-                    }
-                    
-                    tmp.atomSetNode = (*tmp.stageNode).second.atomSet;
-                    while (!tmp.atomSetNode.has_value())
-                    {
-                        tmp.stageNode++;
-                        if (tmp.stageNode == (*tmp.targetNode).second.shaderStageSets.end())
-                            continue;
-                        tmp.atomSetNode = (*tmp.stageNode).second.atomSet;
-                    }
-                    break;
+                    tmp++;
+                    return tmp;
                 }
+                tmp.atomSetNode = std::addressof((*tmp.stageNode).second.atomSet);
```

Comment:
Is there a reason to use std::addressof here? We can just use `&` if CapabilityAtomSet does not provide `operator&`.